services/can_logger/can_logger.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import socket
+import postgres
+import csv
+import os
+
+def write_to_csv(timestamp, can_id, can_data, host_interface):
+
+
+    with open('/data/log/' + host_interface + '.csv', mode = 'a') as log:
```

Comment:
Curious about the performance implications of opening and closing the same file for each can message (many per second) received. While less elegant, a `csv_init()` method that returns a file descriptor that gets passed to `write_to_csv()` might be more performant. Might have to flush the fd here and there?

Reply:
Not sure about implications with numbers, but with something as busy as CAN, it can heavily stress the CPU. My only concern is the moment to flush the file, as the logger is meant to always run, only to stop if an exception arises or the process is terminated.

Reply:
Not sure I follow. Could you explain a little more? Do you mean like the isoblue being unplugged in the middle of a flush?

Reply:
No, I mean, I am not sure when the fd can be flushed as the program is meant to always run and write to file. I understand that the fd should be flushed as soon as the program is done accessing the file, but in this situation, the program would be done only if it is stopped by the user. Other events might not be predictable, as unplugging or the program crashing for unknown reasons. I could be wrong though.